manim/scene/scene.py

```diff
@@ -828,6 +840,65 @@ def update_skipping_status(self):
                 file_writer_config['skip_animations'] = True
                 raise EndSceneEarlyException()
 
+    def handle_caching_play(func): 
+        """
+        This method is used internally to wrap the passed function into a function that will compute the hash of the play invocation, 
+        and will act accordingly : either skip the animation because already cached, either nothing and let the play invocation be processed normally.
+
+        Parameters
+        ----------
+        *args : 
+            Animation or mobject with mobject method and params
+
+        **kwargs : 
+            named parameters affecting what was passed in *args e.g run_time, lag_ratio etc.
```

Comment:
1. Documentation for args and kwargs aren't `*args :` and `**kwargs :`. They are `args : <type of arg expected>` and `kwargs : <type of kwarg expected`. But of course this isn't optimal here, as outlined below.

2. The `Parameters` section lists the parameters for the current function (or for the `__init__` function, in a class). As such, it shouldn't be used here (this lists parameters for the _returned_ function!). I recommend you specify those as text instead, in the `Returns` section, which should specify a type of `Callable[[...], <return type>]` (this Callable is the returned function), and then on the description specify the arguments expected.

Reply:
So I thought to write seomthing like this, but I don't know what to write for the parameters of `scene.play` as it can take a lot of different one :
```python
        func : :class:`Callable[[args : {what do you I put here ?}}], None]`    
            The play() like function that has to be written to the video file stream.
```

Reply:
So, the syntax is `Callable[[pos, arg, types, separated, by, commas], return type]` (also, don’t surround it with :class:\` \`, as Callable is not a class but a generic type for anything that can be called). What we can do is specify “...” as the list of arg types, which is basically saying “anything goes” (but we are more detailed than that in the description).

So:
`Callable[[...], None]`
then in the description specify that it takes the same parameters as :meth:\`play`.

You can also add a “See Also” section and, under it, :meth:\`play\`.

Reply:
Should be fixed now !

Reply:
Nothing was changed...? @huguesdevimeux